### DISCUSSION TITLE
ConnectorSession for Custom Scaler UDF

### DISCUSSION THREAD

**Discussion by blaztoff** [prestodb/presto#18536] [General]
Updating my UDFs from 0.238 to 0.261. 

I have a simple UDF.

```
/**
     * Returns the identity. Mostly for testing.
     *
     * @param session
     *            Connector session, populated by the Presto runtime
     * @return identity, as a string
     */
    @ScalarFunction("print_identity")
    @SqlType(StandardTypes.VARCHAR)
    @Description("Returns the identity represented as a string")
    public static Slice printIdentity(ConnectorSession session) {
        return Slices.utf8Slice(session.getIdentity().toString());
    }
```

 Pre 0.245 the Session would get populated automatically. Most of my UDF's use Session as the first Argument. Now it is forcing me to use a Scaler for All Arguments. And it looks like ConnectorSession is not getting auto populated anymore. What changed here?  How do I get the Session information now?  Trying to look for the documentation on the changes.

Thanks in Advance

**Comment by avsned** [comment]
If only interested in user , we can pull it using SqlFunctionProperties as UDF method param .I also came across similar requirement 
`properties.getSessionUser()`
